src/OpenTK.Mathematics/Geometry/Ray3.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Ray.cs
+//
+// Copyright (C) 2021 OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+
+using System;
+using System.Diagnostics.Contracts;
+
+namespace OpenTK.Mathematics.Geometry {
+
+    /// <summary>
+    /// Defines a ray in 3d space.
+    /// </summary>
+    public struct Ray3 {
+
+        /// <summary>
+        /// Starting point of the ray.
+        /// </summary>
+        public Vector3 Origin { get; set; }
+
+        private Vector3 _direction { get; set; }
+
+        /// <summary>
+        /// Direction of the ray.
+        /// </summary>
+        public Vector3 Direction {
+            get => _direction;
+            set => _direction = value.Normalized();
+        }
+
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) {
+            return obj is Ray3 ray && Equals(ray);
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(Ray3 other) {
+            return Origin.Equals(other.Origin) &&
+                Direction.Equals(other.Direction);
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode() {
+            return HashCode.Combine(Origin, Direction);
+        }
+
+        /// <inheritdoc/>
+        public override string ToString() {
+            return $"{Origin} -> {Direction}";
```

Comment:
This arrow isn't the most intuitive representation for a ray.
I like using the ray equation (`O + tD`), though we can discuss what the best string representation is.

Reply:
I am also not sure if overriding `ToString()` like that makes sense, If you need it for debugging purposes then you could attach `[DebuggerDisplay(...)]`